File: .tmp_py4013/batch_17_dump/3880_minimum_absolute_difference_between_two_values.js
// LeetCode 3880 - Minimum Absolute Difference Between Two Values
// https://leetcode.com/problems/minimum-absolute-difference-between-two-values/

var minAbsoluteDifference = function(nums) {
    const n = nums.length;
    let ans = n + 1;
    const last = [-ans, -ans, -ans];
    for (let i = 0; i < n; i++) {
        const x = nums[i];
        if (x !== 0) {
            ans = Math.min(ans, i - last[3 - x]);
            last[x] = i;
        }
    }
    if (ans > n) return -1;
    return ans;
};

File: .tmp_rs4013/rust_check21/3880_minimum_absolute_difference_between_two_values.rs
struct Solution;
// LeetCode 3880 - Minimum Absolute Difference Between Two Values
// https://leetcode.com/problems/minimum-absolute-difference-between-two-values/

impl Solution {
    pub fn min_absolute_difference(nums: Vec<i32>) -> i32 {
        let n = nums.len() as i32;
        let mut ans = n + 1;
        let mut last = [-ans, -ans, -ans];
        for (i, &x) in nums.iter().enumerate() {
            if x != 0 {
                ans = ans.min(i as i32 - last[(3 - x) as usize]);
                last[x as usize] = i as i32;
            }
        }
        if ans > n {
            -1
        } else {
            ans
        }
    }
}
